batcher/message_batcher.go

```diff
@@ -31,8 +43,13 @@ type batcher struct {
 	largestSeq    *big.Int
 	largestSubSeq int
 
+	// smallestSeq and smallestSubSeq are used to track the highest sequence number
+	// of any record in the batch. This is used for checkpointing.
+	smallestSeq    *big.Int
+	smallestSubSeq int
```

Comment:
I considered that but it didn't seem like a big enough win to be worth changing the interface, imo.  I also briefly considered creating a SequenceToken interface, but that felt a bit over engineered.  Might be worth it we need to make more changes to the batcher.